src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -1878,6 +1878,165 @@ public function get_test_plugin_theme_auto_updates() {
 		return $result;
 	}
 
+	/**
+	 * Test available disk-space for updates/upgrades.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return array The test results.
+	 */
+	public function get_test_available_updates_disk_space() {
+		$disabled              = explode( ',', ini_get( 'disable_functions' ) );
```

Comment:
I think something like this should be enough here:
```
// Sometimes `disk_free_space()` is disabled via the `disable_functions` option for "security purposes".
if ( function_exists( 'disk_free_space' ) ) {
	$available_space = (int) @disk_free_space( WP_CONTENT_DIR . '/upgrade/' );
} else {
	$available_space = false;
}
```
See https://core.trac.wordpress.org/ticket/51857#comment:116 for more context.